crates/revm/src/inspector/instruction.rs

```diff
@@ -1,49 +1,39 @@
+use crate::EVMImpl;
 use alloc::boxed::Box;
 use revm_interpreter::{
     opcode::{BoxedInstruction, Instruction},
     primitives::{db::Database, Spec},
     InstructionResult, Interpreter,
 };
 
-use crate::EVMImpl;
-
 /// Outer closure that calls Inspector for every instruction.
 pub fn inspector_instruction<'a, SPEC: Spec + 'static, DB: Database>(
     instruction: Instruction<EVMImpl<'a, SPEC, DB>>,
 ) -> BoxedInstruction<'a, EVMImpl<'a, SPEC, DB>> {
-    let inspector_instruction = Box::new(
+    Box::new(
         move |interpreter: &mut Interpreter<'_>, host: &mut EVMImpl<'a, SPEC, DB>| {
-            // step
-            let data = &mut host.data;
-            // Safety: as the PC was already incremented we need to subtract 1 to preserve the
-            // old Inspector behavior.
-            unsafe {
-                interpreter.instruction_pointer = interpreter.instruction_pointer.sub(1);
-            }
             if let Some(inspector) = host.inspector.as_mut() {
-                inspector.step(interpreter, data);
+                // Safety: as the PC was already incremented we need to subtract 1 to preserve the
```

Comment:
```suggestion
                // SAFETY: as the PC was already incremented we need to subtract 1 to preserve the
```
According to https://std-dev-guide.rust-lang.org/policy/safety-comments.html#inside-safe-elements